machineconfiguration/v1alpha1/types_internalreleaseimage.go

```diff
@@ -0,0 +1,112 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	machineosconfig "github.com/openshift/api/machineconfiguration/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=internalreleaseimages,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2510
+// +openshift:file-pattern=cvoRunLevel=0000_80,operatorName=machine-config,operatorOrdering=01
+// +openshift:enable:FeatureGate=NoRegistryClusterOperations
```

Comment:
(Not sure on this point) I think this is technically not needed since the alpha API would only be available in techpreview clusters, and we use this generally for GA'ed APIs adding new fields (in this case, in the MCN API). Not sure if there's a downside to having this though.

Reply:
I found it also on the PinnedImageSet though

Reply:
This marker being here helps the generators to know which versions of the resource to create. It will create Custom/DevPreview/TechPreview versions but not a default version, which will prevent this from accidentally being installed on a default cluster